pulp_container/tests/functional/api/test_sign_manifests.py

```diff
@@ -0,0 +1,88 @@
+import os
+import stat
+import pytest
+
+from pulp_smash.pulp3.bindings import monitor_task
+from pulp_smash.utils import execute_pulpcore_python, uuid4
+
+from pulpcore.client.pulp_container import RepositorySign
+
+from pulp_container.tests.functional.constants import REGISTRY_V2_REPO_PULP
+
+
+@pytest.fixture
+def distribution(registry_client, local_registry, container_distribution_api, add_to_cleanup):
+    """The fixture for a distribution that references a repository of the push type."""
+    image_path = f"{REGISTRY_V2_REPO_PULP}:manifest_a"
+    registry_client.pull(image_path)
+    local_registry.tag_and_push(image_path, "test-1:manifest_a")
+
+    distribution = container_distribution_api.list(name="test-1").results[0]
+    add_to_cleanup(container_distribution_api, distribution.pulp_href)
+
+    return distribution
+
+
+@pytest.fixture(scope="session")
+def signing_service(
+    cli_client,
+    gpg_metadata,
+    signing_script_filename,
+    gnupghome_filename,
+    signing_service_api_client,
+):
+    """A fixture for a signing service."""
+    st = os.stat(signing_script_filename)
+    os.chmod(signing_script_filename, st.st_mode | stat.S_IXUSR | stat.S_IXGRP | stat.S_IXOTH)
+
+    gpg, fingerprint, keyid = gpg_metadata
+
+    service_name = uuid4()
+    cmd = (
+        "pulpcore-manager",
+        "add-signing-service",
+        service_name,
+        signing_script_filename,
+        keyid,
+        "--class",
+        "container:ManifestSigningService",
+        "--gnupghome",
+        gpg.gnupghome,
+    )
+
+    with open(gnupghome_filename, "w") as f:
+        f.write(gpg.gnupghome)
+
+    response = cli_client.run(cmd)
+
+    assert response.returncode == 0
+
+    signing_service = signing_service_api_client.list(name=service_name).results[0]
+    assert signing_service.pubkey_fingerprint == fingerprint
+    assert signing_service.public_key == gpg.export_keys(keyid)
+
+    yield signing_service
+
+    cmd = (
+        "from pulpcore.app.models import SigningService;"
+        f"SigningService.objects.filter(name='{service_name}').delete()"
+    )
+    execute_pulpcore_python(cli_client, cmd)
```

Comment:
This shows a gap. Seems like everyone wanted to have a signing service, no one ever wanted to delete one.

Reply:
Do we want to move the creation and removal of signing services to pulpcore as separate fixtures?

Reply:
No, i'm just saying that we provide a command to create one, but not for deletion, which is an asymmetry.